common/metrics/tags.go

```diff
@@ -383,3 +400,50 @@ func DestinationTag(value string) Tag {
 		value: value,
 	}
 }
+
+func VersioningBehaviorBeforeOverrideTag(behavior enumspb.VersioningBehavior) Tag {
+	return &tagImpl{key: behaviorBefore, value: behavior.String()}
+}
+
+func VersioningBehaviorAfterOverrideTag(behavior enumspb.VersioningBehavior) Tag {
+	return &tagImpl{key: behaviorAfter, value: behavior.String()}
+}
+
+// RunInitiatorTag creates a tag indicating how a workflow run was initiated.
+// It handles both new workflow runs and continuations from previous runs.
+// When attributes is nil (e.g. during AddWorkflowExecutionOptionsUpdatedEvent),
+// it returns a tag indicating an existing run.
+func RunInitiatorTag(prevRunID string, attributes *historypb.WorkflowExecutionStartedEventAttributes) Tag {
+	if attributes == nil {
+		return &tagImpl{key: runInitiator, value: existingRun}
+	}
+
+	if prevRunID == "" {
+		return &tagImpl{key: runInitiator, value: newRun}
+	}
+
+	switch attributes.GetInitiator() {
+	case enumspb.CONTINUE_AS_NEW_INITIATOR_WORKFLOW:
+		return &tagImpl{key: runInitiator, value: childRun}
+	case enumspb.CONTINUE_AS_NEW_INITIATOR_RETRY:
+		return &tagImpl{key: runInitiator, value: retryRun}
+	case enumspb.CONTINUE_AS_NEW_INITIATOR_CRON_SCHEDULE:
+		return &tagImpl{key: runInitiator, value: cronRun}
+	default:
+		return &tagImpl{key: runInitiator, value: existingRun}
```

Comment:
probably you want to add a case for `CONTINUE_AS_NEW_INITIATOR_UNSPECIFIED` -> `new` and return `unknown` for default.